src/controller/project_controller.js

```diff
@@ -153,4 +153,50 @@ class ProjectController {
   removeBlockLibrary(blockLibraryName) {
     // TODO: Implement
   }
+
+  /**
+   * Tied to "Export" button. Gets a file name from the user and downloads
+   * the corresponding configuration XML to that file.
```

Comment:
My gut says this should be come two methods.  I suspect there are many variables required, that are not being passed in here, and will differ between the two modes. Fine for now.

That said, the doc comment could use some improvement. Say what it does, first. Probably not necessary to say what UI invokes it; that may change without modifying this method.
"Exports an XML (toolbox or workspace contents) to file, potentially after prompting the user for a file name."